src/main/java/org/elasticsearch/index/query/IdsQueryBuilder.java

```diff
@@ -108,4 +149,84 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     protected String parserName() {
         return IdsQueryParser.NAME;
     }
+
+    public Query toQuery(QueryParseContext parseContext) throws IOException, QueryParsingException {
+        ArrayList<BytesRef> ids = new ArrayList<>();
```

Comment:
can we init the arraylist providing its size as input since we know it? also maybe use List<BytesRef> on the left side?